File: play-core/src/main/kotlin/play/util/collection/ConcurrentLongObjectMap.java
package play.util.collection;

import play.util.function.LongToObjBiFunction;

import javax.annotation.Nullable;
import java.util.function.LongFunction;

public interface ConcurrentLongObjectMap<V> extends Iterable<ConcurrentLongObjectMap.Entry<V>> {

    V get(long key);

    V getOrDefault(long key, V defaultValue);

    V put(long key, V value);

    @Nullable
    V putIfAbsent(long key, V value);

    @Nullable
    V remove(long key);

    boolean remove(long key, V value);

    V computeIfPresent(long key, LongToObjBiFunction<? super V, ? extends V> remappingFunction);

    V computeIfAbsent(long key, LongFunction<? extends V> function);

    V compute(long key, LongToObjBiFunction<? super V, ? extends V> remappingFunction);

    boolean containsKey(long key);

    boolean isEmpty();

    boolean isNotEmpty();

    int size();

    interface Entry<V> {
        long getKey();

        V getValue();
    }
}

File: play-core/src/main/kotlin/play/util/collection/ConcurrentLongObjectMap.kt
package play.util.collection

import play.util.function.LongObjToObjFunction
import play.util.function.LongToObjFunction
import java.util.function.BiFunction
import java.util.stream.LongStream
import java.util.stream.Stream

interface ConcurrentLongObjectMap<V> : Iterable<ConcurrentLongObjectMap.Entry<V>> {
  operator fun get(key: Long): V?
  operator fun set(key: Long, value: V) = put(key, value)
  fun getOrDefault(key: Long, defaultValue: V): V
  fun put(key: Long, value: V): V?
  fun putIfAbsent(key: Long, value: V): V?
  fun remove(key: Long): V?
  fun remove(key: Long, value: V): Boolean
  fun computeIfPresent(key: Long, remappingFunction: LongObjToObjFunction<in V, out V?>): V?
  fun computeIfAbsent(key: Long, function: LongToObjFunction<out V>): V
  fun compute(key: Long, remappingFunction: LongObjToObjFunction<in V?, out V?>): V?
  fun replace(key: Long, oldValue: V, newValue: V): Boolean
  fun merge(key: Long, value: V, remappingFunction: BiFunction<in V?, in V, out V?>): V?
  fun containsKey(key: Long): Boolean
  fun isEmpty(): Boolean
  fun isNotEmpty(): Boolean
  val size: Int
  val keys: LongIterable
  fun keysStream(): LongStream
  val values: Iterable<V>
  fun valuesStream(): Stream<V>
  fun clear()
  interface Entry<V> {
    val key: Long
    val value: V
  }

  companion object {
    @JvmName("create")
    operator fun <V> invoke(): ConcurrentLongObjectMap<V> = ConcurrentLongObjectHashMap()

    @JvmName("create")
    operator fun <V> invoke(initialSize: Int): ConcurrentLongObjectMap<V> = ConcurrentLongObjectHashMap(initialSize)
  }
}
